gnome-initial-setup/gis-page.c

```diff
@@ -445,3 +445,10 @@ gis_page_shown (GisPage *page)
   if (GIS_PAGE_GET_CLASS (page)->shown)
     GIS_PAGE_GET_CLASS (page)->shown (page);
 }
+
+void
+gis_page_reset (GisPage *page)
```

Comment:
It does not feel like this generic method is actually needed here. You could simply add a method to the password page and call it from the account page directly, but I wonder if you could not simply do the cleanup from a visibility signal handler in the password page. The account page already calls `set_password_page_visibility()`...

Reply:
The problem is that all that `set_password_page_visibility()` does is to call `gtk_widget_set_visible()` over the password page, which is fine because it's a `GtkWidget` after all, but the account page can't access any of the internal data that needs to be reset in there, that is, password entry, password confirmation entry, reminder text and the "show password" checkbox.

Those operations deal with internal properties of the password page that need to be dealt with internally (unless we break that encapsulation, but I don't think that's worth it.)